agent.py

```diff
@@ -52,6 +55,28 @@ def pruneBuffer(self):
 		for key in buffer:
 			if (key > old_threshold):
 				buffer.pop(key)
+	
+	def tryPopBuffer(self):
+		#pops buffer and returns the story that is popped
+		#else, returns gracefully
+
+		if (len(buffer) > 0):
+			firstKey = sys.float_info.max
+			for key in buffer:
```

Comment:
When writing code, be wary of for loops! We only want to use for loops when absolutely necessary. Since we have to look for the first key, let's try using a priority queue instead of a dictionary. heapq is a python implementation of priority queue: https://towardsdatascience.com/introduction-to-priority-queues-in-python-83664d3178c3 

An alternative is using a list and pushing elements into the list (like an array), so elements are always in order and we can first-in-first-out. (Lists are always the cheapest way of storing multiple items together in python, by design). However, we may want to change this in the future so that instead of taking the first one we could take the last one, or whatever, so we don't need to optimize this quite yet.  ...Besides, we're already using Python, so obviously we don't care that much about code efficiency as long is it's not horrible --- Python is orders of magnitude slower than C, for example.

Reply:
Just to clarify: you would like for me to completely change the dictionary "buffer" into a priority queue and apply changes to related methods? 
(if so, I'm a little lost on how I can include the timestamps and remove elements when they're too old)